src/gstring.rs

```diff
@@ -206,9 +206,9 @@ impl From<Box<str>> for GString {
     }
 }
 
-impl From<&'static str> for GString {
+impl<'a> From<&'a str> for GString {
     #[inline]
-    fn from(s: &'static str) -> Self {
+    fn from(s: &'a str) -> Self {
         GString::Borrowed(s.as_ptr() as *const c_char, s.len())
     }
 }
```

Comment:
This is wrong, as we outlive `'a` for sure.
Can you add other From with `.to_owned()` conversion instead, if rust allow?

Reply:
In addition to `impl From<&'static str> for GString`? The conflicting impl isn't allowed.

Reply:
How did that work without copying before anyway. We just missed a mistake here before :) We would have to copy it anyway because we need to add a `'\0'` at the end.

Reply:
Alright, PR updated accordingly (I think 😅)